test/extended/authorization/tolerations_restriction.go

```diff
@@ -0,0 +1,136 @@
+package authorization
+
+import (
+	"fmt"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+)
+
+var _ = g.Describe("[Feature: Toleration restrictions] Tolerations restrictions should be functional", func() {
+	defer g.GinkgoRecover()
+	f := e2e.NewDefaultFramework("tolerations-restriction")
+	cs, err := e2e.LoadClientset()
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	g.Describe("Pod with service account", func() {
```

Comment:
grant permission to a named toleration and make sure that passes, another fails, and tolerating everything fails.